### DISCUSSION TITLE
Playing buffered audio data

### DISCUSSION THREAD

**Discussion by GCSBOSS** [bluefireteam/audioplayers#1575] [General]
Does this package support playing buffered audio data?

Basically my process is as follows:

1. Fetch multiple encrypted audio segments from internet one by one
2. Decrypt each file and store all of them in a buffer
3. Need to play whatever is in the buffer and move around pause etc...

Is this supported?

**Comment by Gustl22** [comment]
Only for Android at the moment: see #1269

https://github.com/bluefireteam/audioplayers/blob/main/feature_parity_table.md#main-features

**Reply by GCSBOSS** [reply]
Is it being worked on for other targets already or not planned yet?

**Reply by Gustl22** [reply]
One was working on a Windows implementation, but not actively I guess. I do this in my spare time, so as this is not prioritized right now, I do not work on it. You're most welcome to provide implementations for the other platforms!

**Comment by projectitis** [comment]
I'm guessing this feature would also allow playing generated audio - for example, synth sounds?
My ambition is to build/port a mod player, but I'd need at least support to pump audio data into a (ring)buffer.

**Reply by Gustl22** [reply]
Feel free to start a Merge Request :)
